framework/base/src/storage/mappers/set_mapper.rs

```diff
@@ -174,15 +228,33 @@ where
         self.queue_mapper.iter()
     }
 
-    pub fn iter_from(&self, value: &T) -> Iter<SA, T> {
+    pub fn iter_from_address(&self, address: &ManagedAddress<SA>) -> Iter<SA, T> {
+        self.queue_mapper.iter_from_address(address)
+    }
+
+    pub fn iter_from_value(&self, value: &T) -> Iter<SA, T> {
         let node_id = self.get_node_id(value);
         self.queue_mapper.iter_from_node_id(node_id)
     }
 
+    pub fn iter_from_value_at_address(
+        &self,
+        address: &ManagedAddress<SA>,
+        value: &T,
+    ) -> Iter<SA, T> {
+        let node_id = self.get_node_id_from_address(address, value);
+        self.queue_mapper.iter_from_node_id(node_id)
```

Comment:
I think here we need a new addition in the queue_mapper.
`iter_from_node_id_at_address`